packages/connectors/connector-dingtalk-web/src/types.ts

```diff
@@ -0,0 +1,33 @@
+import { z } from 'zod';
+
+export const dingtalkConfigGuard = z.object({
+  clientId: z.string(),
+  clientSecret: z.string(),
+  scope: z.string().optional(),
+});
+
+export type DingtalkConfig = z.infer<typeof dingtalkConfigGuard>;
+
+export const accessTokenResponseGuard = z.object({
+  accessToken: z.string().optional(),
+  refreshToken: z.string().optional(),
+  expireIn: z.number().optional(), // In seconds
+  corpId: z.string().optional(),
+});
+
+// https://open.dingtalk.com/document/isvapp/dingtalk-retrieve-user-information
+export const userInfoResponseGuard = z.object({
+  nick: z.string().optional(),
+  avatarUrl: z.string().optional(),
+  mobile: z.string().optional(),
+  openId: z.string().optional(), // DingTalk no longer recommends using OpenId for integration. Instead, use unionId.
```

Comment:
Any reference for this conclusion?

Reply:
![CleanShot 2024-05-28 at 14 58 06@2x](https://github.com/logto-io/logto/assets/48710760/74fbf911-6682-494e-b36d-4a32077144c1)